### ISSUE TITLE
Introduction of agent6 causes a failure

### ISSUE THREAD

**Issue by viraptor** [DataDog/puppet-datadog-agent#380]
After updating to a module with agent6 support, I get the following during a build:
```
Notice: /Stage[main]/Datadog_agent::Ubuntu/File[/etc/apt/sources.list.d/datadog.list]/ensure: defined content as '{md5}29206c758f2581d5b0466bdb5ca7a12e'
Info: /Stage[main]/Datadog_agent::Ubuntu/File[/etc/apt/sources.list.d/datadog.list]: Scheduling refresh of Exec[datadog_apt-get_remove_agent6]
Info: /Stage[main]/Datadog_agent::Ubuntu/File[/etc/apt/sources.list.d/datadog.list]: Scheduling refresh of Exec[datadog_apt-get_update]
Notice: /Stage[main]/Datadog_agent::Ubuntu/Exec[datadog_apt-get_remove_agent6]/returns: Reading package lists...
Error: /usr/bin/apt-get remove -y -q datadog-agent returned 100 instead of one of [0]
Error: /Stage[main]/Datadog_agent::Ubuntu/Exec[datadog_apt-get_remove_agent6]/returns: change from notrun to 0 failed: /usr/bin/apt-get remove -y -q datadog-agent returned 100 instead of one of [0]
Notice: /Stage[main]/Datadog_agent::Ubuntu/Exec[datadog_apt-get_remove_agent6]/returns: Building dependency tree...
Notice: /Stage[main]/Datadog_agent::Ubuntu/Exec[datadog_apt-get_remove_agent6]/returns: Reading state information...
Notice: /Stage[main]/Datadog_agent::Ubuntu/Exec[datadog_apt-get_remove_agent6]/returns: W: Duplicate sources.list entry http://security.ubuntu.com/ubuntu/ trusty-security/main amd64 Packages (/var/lib/apt/lists/security.ubuntu.com_ubuntu_dists_trusty-security_main_binary-amd64_Packages)
Notice: /Stage[main]/Datadog_agent::Ubuntu/Exec[datadog_apt-get_remove_agent6]/returns: W: Duplicate sources.list entry http://security.ubuntu.com/ubuntu/ trusty-security/universe amd64 Packages (/var/lib/apt/lists/security.ubuntu.com_ubuntu_dists_trusty-security_universe_binary-amd64_Packages)
Notice: /Stage[main]/Datadog_agent::Ubuntu/Exec[datadog_apt-get_remove_agent6]/returns: W: You may want to run apt-get update to correct these problems
Notice: /Stage[main]/Datadog_agent::Ubuntu/Exec[datadog_apt-get_remove_agent6]/returns: E: Unable to locate package datadog-agent
Notice: /Stage[main]/Datadog_agent::Ubuntu/Exec[datadog_apt-get_remove_agent6]/returns: Reading package lists...
Notice: /Stage[main]/Datadog_agent::Ubuntu/Exec[datadog_apt-get_remove_agent6]/returns: Building dependency tree...
Notice: /Stage[main]/Datadog_agent::Ubuntu/Exec[datadog_apt-get_remove_agent6]/returns: Reading state information...
Error: /Stage[main]/Datadog_agent::Ubuntu/Exec[datadog_apt-get_remove_agent6]: Failed to call refresh: /usr/bin/apt-get remove -y -q datadog-agent returned 100 instead of one of [0]
Notice: /Stage[main]/Datadog_agent::Ubuntu/Exec[datadog_apt-get_remove_agent6]/returns: E: Unable to locate package datadog-agent
Error: /Stage[main]/Datadog_agent::Ubuntu/Exec[datadog_apt-get_remove_agent6]: /usr/bin/apt-get remove -y -q datadog-agent returned 100 instead of one of [0]
```
Happy to provide other details if needed.

I'm not sure why it tries to remove any agent before installing one...

**Comment by truthbk** [comment]
Thanks for the report @viraptor, I will try to debug when we get back to the office. I did not hit anything like this when testing, so it might take some time to reproduce. Could you provide some information regarding your environment (linux distro/version, puppet version, etc)?

Happy New Year! :)

**Comment by viraptor** [comment]
I experience the issue on Ubuntu 14.04 in an EC2 environment. I'm using puppet 3.4.
I think it can be reproduced in vagrant, but I'll retest and let you know.

Happy new year!

**Comment by scottgeary** [comment]
I think this is the same bug: https://github.com/DataDog/puppet-datadog-agent/pull/384

The beta fact `::apt_agent6_beta_repo` getting parsed as a string, as causing all sorts of chaos.

**Comment by truthbk** [comment]
Working at the moment in the 2.0 release, that module will not support EOL'd puppets and _will_ include @scottgeary's fix in #384 (even though it might actually not affect newer puppets, will have to verify). I will also shortly after also draft a `1.x` release that will backport the same fix. Should be happening in the next few days.

**Comment by truthbk** [comment]
Closing after release of 2.0 where I _believe_ this was fixed and where #384 was indeed included. We can reopen if anything weird comes up to this regard. Thank you @viraptor, @scottgeary!
